\documentclass[a4paper, 12pt]{article}

% basic package list
%\usepackage[T1]{fontenc}
\usepackage{fontspec}
\defaultfontfeatures{Mapping=tex-text}
\usepackage[margin=25mm]{geometry}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}

% other packages
\usepackage{xunicode}
\usepackage{xltxtra}
\usepackage{hyperref}         % hyperlinks
\usepackage{booktabs}         % professional-quality tables
\usepackage{indentfirst}      % to indent section first paragraph
\usepackage{url}              % simple URL typesetting
\usepackage{natbib}
\usepackage[modulo]{lineno}
\usepackage{sectsty}          % to change section font size
\usepackage[flushleft]{threeparttable} % table with note
\usepackage{mathtools}
\usepackage{indentfirst}
\usepackage{multirow}
\usepackage[labelfont=bf]{caption}
\usepackage{lineno}
\usepackage{courier}
\usepackage{rotating} % to rotate table vertically (for table S2)
%\usepackage{tablefootnote} % to add table footnote


% black hypelinks with no border
\hypersetup{
    colorlinks,
    citecolor=black,
    filecolor=black,
    linkcolor=black,
    urlcolor=black
}

% set additional parameters
\setcitestyle{authoryear,open={(},close={)}}
\graphicspath {{Figures/}}

\sectionfont{\fontsize{14}{19}\selectfont}
\subsectionfont{\fontsize{12}{17}\selectfont}

\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}

\setcounter{figure}{0}

% reduce page number font size
\renewcommand*{\thepage}{\footnotesize\arabic{page}}

\title{Supplementary Material for: Joint inference of adaptive and demographic history from temporal population genomic data}
\author{\small
            Vitor A. C. Pavinato$^{1,2,3}$, Stéphane De Mita$^4$, Jean-Michel Marin$^2$, Miguel de Navascués$^{1,5*}$}
\date{{\footnotesize %
    $^1$CBGP, INRAE, CIRAD, IRD, Montpellier SupAgro, Université de Montpellier, Montpellier, France\\%
    $^2$UMR Institut Montpelliérain Alexander Grothendieck, Université de Montpellier, France\\%  
    $^3$Entomology Dept., CFAES, The Ohio State University, Wooster, USA\\%
    $^4$UMR Interactions Arbres-Microorganismes, INRAE, France \\%
    $^5$Human Evolution, Department of Organismal Biology, Uppsala University, Uppsala, Sweden\\%  
    $^*$corresponding author: miguel.navascues@inrae.fr\\[2ex]%
    }
    \footnotesize\today    
}
\begin{document}
\maketitle

\newpage

\section*{S1 Supplementary Table}

\begin{table}[ht]
 \caption{\textbf{Notation.}}
  \centering
  \label{table:tableS1}
  \begin{tabular}{ll}
   \cmidrule(r){1-2}
    Symbol                  & Meaning \\
    \midrule
    $r$        & recombination rate per base pair per generation \\
    $G$                     & genome size\\
    $h$                     & dominance coefficient\\
    $L$                     & mean substitution load\\
    $N$                     & population size during the inference period\\ 
    $N_{\mathrm{0}}$        & population size during the burn-in period\\
    $N_{\mathrm{e}}$        & effective population size during the inference period\\
    $P_{\mathrm{R}}$        & proportion of non-neutral regions\\
    $P_{\mathrm{B}}$        & proportion of beneficial mutations per non-neutral region\\
    $P$                     & proportion of polymorphisms under strong selection, i.e. $s > 1/ N_{\mathrm{e}}$\\
    $s$                     & selection coefficient\\
    $\bar{s}$               & average selection coefficients of mutations under strong selection\\
    $t_1$ and $t_2$         & generations were samples of individuals were taken\\
    $W_{\mathrm{max}i}$     & fitness of individual with highest fitness in the population in generation $i$\\
    $\bar{W_{i}}$           & mean fitness of the population in generation $i$\\
    $\gamma$                & mean of the gamma distribution where the selection coefficients of each\\
                            & beneficial mutation was sampled\\
    $\mu$                   & mutation rate per generation\\
    $\mu_{\mathrm{b}}$      & mutation rate of beneficial mutations per generation\\
    $\tau$                  & length of time-interval between the first and the second sample\\
    $\theta_{\mathrm{b}}$   & scaled mutation rate of beneficial mutations\\
&\\
    
    \bottomrule
  \end{tabular}
  \label{tab:supple_symbols}
\end{table}

\section*{S2 Supplementary Methods}

\subsection*{S2.1 List of summary statistics}

Three groups of summary statistics were calculated for the reference table (\texttt{e.g} training data): 1) summary statistics calculated SNP-by-SNPs (i.e. locus-specific summary statistics - LSS); 2) summary statistics calculated for nucleotide sequence (i.e. window summary statistics - WSS) - in this case a window containing base positions was defined around each SNP in the dataset; and 3) summary statistics calculated globally by averaging the locus-specific or window-specific summary statistics (global summary statistics - GSS). Summary statistics informative about within-population diversity as $H_{\mathrm{E}}$ were calculated for each time point individually and marginally by pooling all individuals sampled together. For the summary statistics calculated in a window, three window sizes were used: 500bp, 5Kbp, and 10Kbp. For the site-frequency spectrum, the allele counts were used instead of frequency, and all bins were included in the reference table.

\begin{enumerate}
	\item LSSs - Locus-specific summary statistics:
    \begin{enumerate}
    	\item $D_{\mathrm{j}}$ - Jost's $D$ \citep{Jost:2008cs,Jost:2009hn};
        \item $F_{\mathrm{ST}}$ - Weir and Cockerham's $F_{\mathrm{ST}}$ \citep{Weir:1984dx};
        \item $H_{\mathrm{E}}$ - Expected Heterozygosity;
    \end{enumerate}
    \item WSSs - Window-specific summary statistics (window sizes - 500bp, 5Kbp and 10Kbp):
    \begin{enumerate}
        \item $D_{\mathrm{j}}$ - Jost's $D$;
        \item $F_{\mathrm{ST}}$ - Weir and Cockerham's $F_{\mathrm{ST}}$;
        \item $D_{\mathrm{a}}$ - Net distance between populations;
        \item $H_{\mathrm{E}}$ - Expected Heterozygosity;
        \item $S$ - Number of polymorphic sites;
        \item $\pi$ - Nucleotide diversity \citep{Nei:1979uk};
        \item $\theta_{\mathrm{W}}$ - Watterson's 4Nu estimator \citep{Watterson:1975bh};
        \item $T_{\mathrm{j}}D$ - Tajima's D \citep{Tajima:1989un};
    \end{enumerate}
    \item Global summary statistics:
    \begin{enumerate}
        \item All summary statistics enumerated above (including the within population calculations);
        \item SFS bins - binned Joint Site-frequency spectrum $SFS$ \citep{Ewing:2016gv};
        \item mean, variance, kurtosis, skewness and the 5\% and 95\% quantiles of all intra-locus summary statistics presented above (LSSs and WSSs);
    \end{enumerate}
\end{enumerate}

\section*{S3 Supplementary Results}

\subsection*{S3.1 ABC-RF}

Figure~\ref{fig:supple_pods_priors} shows the prior distribution of the parameters and latent variables that were estimated with the ABC-RF. These distributions correspond to the values that were used to train ABC-Random forests. 

Figure~\ref{fig:supple_oob_sel} shows the distribution of Random Forest OOB estimates on the observed values (the ones used to train the RF) for model parameters and other latent variables informative about selection. The ABC-RF regressions performed poorly on the inference of two parameters of the selection dynamics $P_{\mathrm{R}}P_{\mathrm{B}}$ and $\gamma$. On the other hand, the ABC-RF performed better on the latent variables $P$, and $\bar{s}$ (which can be viewed as the realization of the two aforementioned parameters). Each value of these latent variables was obtained from the simulation by 1) calculating the relative proportion of strongly beneficial mutations (those with $s > 1/N_{\mathrm{e}}$) over all segregating mutations, 2) averaging their $s$ values. 
The estimation of substitution load $L$ had also somehow good performance. It performed well for higher values but poorly for lower values, as we can see in the distribution of the estimated OOB values against the true values (Figure~\ref{fig:supple_oob_sel} e). It seems that only strong adaptive dynamics (many loci under selection or strong selection) leaves a polymoprhism patterns that are informative about substitution load. The proportion of strongly beneficial mutations that segregated in the population, $P$ can also be used as a complementary estimator to characterize the adaptive dynamics (it is complementary to $\theta_{\mathrm{b}}$ presented in the main text). 

Figure~\ref{fig:supple_oob_demo} also presents the distribution OOB estimates with the true observed values for model parameters and for the latent variables informative about drift. For these model parameters, ABC-RF could not satisfactory recover the recombination rate $r$ as expected given the summary statistics calculated and the information contained in the temporal allele frequency dynamics. On the other hand, the ABC-RF estimated the mutation rate per nucleotide position per generation $\mu$, satisfactorily. The last latent variable also performed relatively well and similarly to the $L$, and could be used in combination with other estimates to characterize the selection/drift dynamics, when selection is more frequent. 

Figure~\ref{fig:supple_pods_varplots_sel} and Figure~\ref{fig:supple_pods_varplots_demo} show the variable importance plots (VIP) of each ABC-RF that were grown to infer the parameters of our ABC-RF framework. In a nutshell, the VIP shows how many times a summary statistics was selected by the RF during the training phase. Consequently, the VIP shows the relative importance of each summary statistics for the training of that parameter (more details in the main text). For the latent variables that account for the selection and drift dynamics, the top summary statistics were those calculated from the genome-wide distribution of locus-specific summary statistics that reflect their genome-wide distribution - variance, mean, skewness, kurtosis and quantiles. For example, for the latent variables $P$ and $\bar{s}$, RF used more frequently summary statistics informative about the distribution of locus-specific heterozygosity calculated globally and calculated for the second sample, and summary statistics that inform about the distribution of genome-wide intra-locus $F_{\mathrm{ST}}$. For $L$ summary statistics informative about the genetic divergence between samples: $F_{\mathrm{ST}}$, Da and Dj were more frequently selected. 

\subsection*{S3.3 Analysis of temporal genomic data of feral populations of \textit{Apis mellifera}}

Figure~\ref{fig:supple_feralbee_thetab}, Figure~\ref{fig:supple_feralbee_N}, and Figure~\ref{fig:supple_feralbee_NE} show the distribution of the true and the OOB estimated values for the parameters used in this work for the joint inference of demography and selection for temporal pairs of populations of feral \textit{A. mellifera}. The OOB MSE and $R^2$ estimates showed that even with fewer simulations for model training, the RF grew for the  \textit{A. mellifera} data had comparable performance to the RF grew for the simulated data.

The performance of the model parameters and additional latent variables for all temporal pairs of this dataset also were similar to the simulations used for the framework evaluation. Figure~\ref{fig:supple_feralbee_prpb}, Figure~\ref{fig:supple_feralbee_pstrong}, Figure~\ref{fig:supple_feralbee_gammamean}, Figure~\ref{fig:supple_feralbee_gammaselmean}, and Figure~\ref{fig:supple_feralbee_load} show the parameters and latent variables informative about the adaptive history. Figure~\ref{fig:supple_feralbee_mu}, Figure~\ref{fig:supple_feralbee_r}, and Figure~\ref{fig:supple_feralbee_nen} show the model parameters and an additional latent variable complementary to the mutation load for the inference of the selection impact on demography.

Table~\ref{tab:supple_bees_posteriors} contains the posterior estimates of each parameter for each \textit{A. mellifera} population. The last row shows the $F_{\mathrm{ST}}$-NE estimates for comparison. Figure~\ref{fig:supple_feralbee_densityselection} and Figure~\ref{fig:supple_feralbee_densitydemo} show the prior and posterior distributions of all remaining model parameters and latent variables calculated for the temporal samples analyzed with ABC-RF framework.\\

% Table Supplementary  2 - Posterior estimates parameter/population.
 \begin{sidewaystable}[!ht]
 \begin{small}
 \caption{\textbf{\small Parameter posterior median and 95\% credibility interval (in parenthesis) for each  \textit{A. mellifera} feral population.}}
 \end{small}
   \begin{tiny}
    \centering
    \begin{tabular}{cccccccc}
    \cmidrule(r){1-8}
        Parameter & Avalon & Humboldt & Davis & Stanislaus & Stebbins & Riverside & Placerita \\
        \midrule
        $P_\mathrm{R}P_\mathrm{B}$  & 1.53E-04 & 8.35E-04 & 1.26E-04 & 2.39E-04 & 9.70E-05 & 8.24E-05 & 6.19E-04 \\
         & (1.59E-07 – 2.48E-01) & (1.55E-06 – 2.02E-01) & (1.17E-07 – 4.99E-02) & (4.32E-06 – 1.70E-01) & (1.14E-07 – 4.64E-02) & (1.18E-07 – 5.45E-01) & (4.15E-07 – 3.54E-01) \\
        $P$ & 1.48E-05 & 4.12E-05 & 1.89E-05 & 3.26E-05 & 2.61E-05 & 3.26E-05 & 9.39E-05 \\
         & (1.45E-06 – 1.04E-03) & (8.58E-06 – 3.08E-03) & (1.82E-06 – 3.99E-04) & (2.16E-06 – 3.49E-04) & (2.95E-06 – 5.23E-04) & (3.32E-06 – 4.93E-01) & (4.90E-06 – 4.93E-01) \\
        $\gamma$ & 1.77E-02 & 2.67E-02 & 2.14E-02 & 2.29E-02 & 2.35E-02 & 3.20E-02 & 2.67E-02 \\
         & (1.12E-03 – 8.82E-01) & (1.22E-03 – 8.45E-01) & (1.12E-03 – 9.12E-01) & (1.05E-03 – 8.93E-01) & (1.04E-03 – 8.93E-01) & (1.03E-03 – 8.93E-01) & (1.15E-03 – 8.92E-01) \\
        $\bar{s}$ & 9.72E-02 & 3.07E-01 & 1.75E-01 & 2.30E-01 & 1.58E-01 & 2.19E-01 & 3.31E-01 \\
         & (4.31E-03 – 1.62E+00) & (3.22E-02 – 2.23E+00) & (1.95E-03 – 1.50E+00) & (1.95E-03 – 1.30E+00) & (1.31E-03 – 1.51E+00) & (1.46E-03 – 2.25E+00) & (6.17E-03 – 1.98E+00) \\
        $\theta_{\mathrm{b}}$ & 3.72E-01 & 1.73E+00 & 2.90E+00 & 3.46E+00 & 3.07E+00 & 2.90E+00 & 6.22E+00 \\
         & (1.72E-04 – 5.66E+03) & (1.22E-03 – 1.56E+04) & (1.70E-02 – 1.32E+03) & (2.66E-02 – 1.37E+03) & (9.20E-03 – 4.61E+02) & (4.41E-03 – 2.04E+04) & (1.30E-02 – 1.70E+04) \\
        $L$ & 1.01E-03 & 1.59E-04 & 2.06E-01 & 2.65E-01 & 2.16E-04 & 9.43E-03 & 2.59E-02 \\
         & (1.14E-09 – 9.64E-01) & (1.42E-16 – 9.27E-01) & (5.32E-08 – 8.29E-01) & (1.22E-06 – 8.78E-01) & (2.00E-16 – 4.62E-01) & (1.47E-08 – 8.71E-01) & (2.01E-16 – 7.56E-01) \\
        $\mu$ & 5.67E-08 & 8.05E-08 & 5.03E-08 & 2.56E-08 & 7.12E-08 & 2.98E-08 & 1.11E-07 \\
         & (3.45E-09 – 1.65E-07) & (9.75E-09 – 1.66E-07) & (4.04E-09 – 1.57E-07) & (3.06E-09 – 1.45E-07) & (4.86E-09 – 1.66E-07) & (2.71E-09 – 1.65E-07) & (1.23E-08 – 1.67E-07) \\
        $r$ & 3.72E-08 & 5.19E-08 & 2.71E-08 & 3.19E-08 & 2.28E-08 & 2.10E-08 & 3.54E-08 \\
         & (1.02E-08 – 6.14E-06) & (1.02E-08 – 2.06E-06) & (1.02E-08 – 2.01E-06) & (1.04E-08 – 1.06E-06) & (1.02E-08 – 8.54E-07) & (1.02E-08 – 1.06E-06) & (1.02E-08 – 3.64E-06) \\
        $N$ & 8.10E+01 & 4.90E+01 & 1.02E+03 & 1.80E+03 & 3.42E+02 & 5.19E+02 & 1.39E+02 \\
         & (1.70E+01 – 5.36E+03) & (1.00E+01 – 7.71E+03) & (2.42E+02 – 6.78E+03) & (3.67E+02 – 8.00E+03) & (1.58E+02 – 3.09E+03) & (6.30E+01 – 8.72E+03) & (1.40E+01 – 4.69E+03) \\
        $N_{\mathrm{e}}$ & 8.30E+01 & 4.42E+01 & 1.04E+03 & 1.96E+03 & 3.80E+02 & 5.69E+02 & 1.46E+02 \\
         & (2.21E+01 – 9.59E+02) & (4.55E+00 – 5.76E+02) & (2.03E+02 – 7.24E+03) & (4.19E+02 – 8.70E+03) & (1.76E+02 – 3.90E+03) & (5.53E+01 – 8.17E+03) & (4.57E+00 – 1.15E+03) \\
        $N_{\mathrm{e}}/N$ & 9.07E-01 & 1.05E+00 & 1.07E+00 & 1.06E+00 & 1.09E+00 & 1.04E+00 & 1.06E+00 \\
         & (1.15E-03 – 1.13E+00) & (8.75E-04 – 1.17E+00) & (5.63E-01 – 1.12E+00) & (5.26E-01 – 1.12E+00) & (2.28E-01 – 1.13E+00) & (2.78E-01 – 1.13E+00) & (2.44E-01 – 1.14E+00) \\
         \midrule
        $F_{\mathrm{ST}}$-NE & 8.00E+01 & 1.13E+02 & 3.62E+01 & 2.28E+01 & 3.60E+01 & 1.47E+01 & 6.15E+01 \\
         & (0.00E+00 – 7.80E+02) & (5.30E+00 – 5.88E+17) & (8.63E+00 – 3.59E+02) & (6.00E+00 – 3.42E+02) & (5.63E+00 – 2.25E+02) & (1.79E+00 – 1.20E+02) & (4.46E+00 – 6.12E+02) \\
          \bottomrule
    \multicolumn{8}{l}{\tiny  Estimates are in the natural scale.} \\
    \end{tabular}
\label{tab:supple_bees_posteriors}
\end{tiny}
\end{sidewaystable}



\newpage
\bibliographystyle{apalike}
\bibliography{references}

% Add all supplementary figure here:
\newpage

% Figure Supplementary 1 - Prior distribution of the PODs
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.85\textwidth]{Figures/FigureS1_parameters_histograms.pdf}
  \small\caption{\textbf{Prior distribution of the simulation parameters, latent variables informative about the adaptive history and demography inferred with the ABC-RF.} (a) the per site mutation rate $\mu$, (b)  per base recombination rate per generation $r$, (c) the population size $N$, (d) the probability of beneficial mutation $P_RP_B$, (e) mean of the gamma distribution $\gamma$, (f) the proportion of strongly selected mutations $P$, (g) average selection coefficients of mutations under strong selection $\bar{s}$, (h) the scale mutation rate of beneficial mutations $\theta_{\mathrm{b}}$, (i) the average substitution load $L$, (j) the effective population size $N_{\mathrm{e}}$, (k) the distance between the effective size and population size expressed in the ratio $N_{\mathrm{e}}/N$.}
  \label{fig:supple_pods_priors}
\end{figure}

% Figure Supplementary 2 - PODs OOB plots << selection >>
\begin{figure}[ht]
  \centering
  \includegraphics[width=1\textwidth]{Figures/FigureS2_oob_plots_selection.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of model parameters informative about the number of beneficial mutations and their strength, the latent variables associated with these two parameters, and variables informative about the adaptive dynamic.} (a) probability of a beneficial mutation to arise $P_RP_B$; (b) number of mutations under strong selection $P$; (c)  mean of the gamma distribution $\gamma$; (d) mean of selection coefficients of mutations under strong selection $\bar{s}$; and (e) mean substitution load $L$. The barplots on the left side of the scatterplots of $P$, $\bar{s}$, and $L$ correspond to the posterior estimates obtained for the simulations removed for the training of these parameters, as these simulations had zero as true values. MSE and $R^2$ estimates of b, d, and e corresponded to non-neutral PODs. See Table S2, Supplementary results, for MSE and $R^2$ estimates for neutral PODs.}
  \label{fig:supple_oob_sel}
\end{figure}

% Figure Supplementary 3 - PODs OOB plots << demography >>
\begin{figure}[ht]
  \centering
  \includegraphics[width=1\textwidth]{Figures/FigureS3_oob_plots_demography.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of model parameters and latent variables informative about demography.} (a) mutation rate per generation $\mu$; (b) per base recombination rate per generation $r$; and (c) the ratio between the effective size and the population census size $N_{\mathrm{e}}/N$.}\label{fig:supple_oob_demo}
\end{figure}

% Figure Supplementary 4 - Variable importance plots - PODs selection parameters
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS4_varplot_selection.pdf}
  \small\caption{\textbf{Variable importance plots of the trained ABC-RF for parameters informative about the adaptive history.} (a) The probability of beneficial mutation $P_RP_B$, (b) the proportion of strongly selected mutations $P$, (c) mean of the gamma distribution $\gamma$, (d) average selection coefficients of mutations under strong selection $\bar{s}$, (e) the scale mutation rate of selected mutations $\theta_{\mathrm{b}}$, and (f) the average substitution load $L$.}
  \label{fig:supple_pods_varplots_sel}
\end{figure}

% Figure Supplementary 5 - Variable importance plots - PODs demography parameters
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS5_varplot_demography.pdf}
  \small\caption{\textbf{Variable importance plots of the trained ABC-RF for parameters informative about the adaptive history.} (a) the per site mutation rate $\mu$, (b) per base recombination rate per generation $r$, (c) the population size $N$, (d) the effective population size $N_{\mathrm{e}}$, (e) the distance between the effective size and population size expressed in the ratio $N_{\mathrm{e}}/N$.}
  \label{fig:supple_pods_varplots_demo}
\end{figure}

% Figure Supplementary 6 - Expected vs Predicted for simulations with variable recombination rate
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS6_posterior_plots_recombination.pdf}
  \small\caption{\textbf{Expected versus posterior estimates plots for PODs with variable genome-wide recombination rates.} (a) mutation rate per generation $\mu$, (b) per base recombination rate per generation $r$, (c) the population size $N$, (d) the effective population size $N_{\mathrm{e}}$, (e) the ratio between the effective size and the population census size $N_{\mathrm{e}}/N$, (f) The probability of beneficial mutation $P_RP_B$, (g) the proportion of strongly selected mutations $P$, (h) mean of the gamma distribution $\gamma$, (i) average selection coefficients of mutations under strong selection $\bar{s}$, (j) the scale mutation rate of selected mutations $\theta_{\mathrm{b}}$, and (k) the average substitution load $L$. MSE and $R^2$ estimates of g, i and k corresponded to non-neutral PODs. See Table S2, Supplementary results, for MSE and $R^2$ estimates for neutral PODs.}
  \label{fig:supple_pods_variable_recomb}
\end{figure}

% Figures Supplementary 7 to 13 Model check PCA.
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS7_model_check_Avalon.pdf}
  \small\caption{\textbf{Prior evaluation for the Avalon population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) If the model captured most of the targeted data features.}
  \label{fig:supple_model_check_avalon}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS8_model_check_Humboldt.pdf}
  \small\caption{\textbf{Prior evaluation for the Humboldt population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) If the model captured most of the targeted data features.}
  \label{fig:supple_model_check_humboldt}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS9_model_check_Davis.pdf}
  \small\caption{\textbf{Prior evaluation for the Davis population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) If the model captured most of the targeted data features.}
  \label{fig:supple_model_check_davis}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS10_model_check_Stanislaus.pdf}
  \small\caption{\textbf{Prior evaluation for the Stanislaus population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) If the model captured most of the targeted data features.}
  \label{fig:supple_model_check_stanislaus}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS11_model_check_Stebbins.pdf}
  \small\caption{\textbf{Prior evaluation for the Stebbins population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) If the model captured most of the targeted data features.}
  \label{fig:supple_model_check_stebbins}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS12_model_check_Riverside.pdf}
  \small\caption{\textbf{Prior evaluation for the Riverside population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) If the model captured most of the targeted data features.}
  \label{fig:supple_model_check_riverside}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS13_model_check_Placerita.pdf}
  \small\caption{\textbf{Prior evaluation for the Placerita population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) If the model captured most of the targeted data features.}
  \label{fig:supple_model_check_placerita}
\end{figure}

% Figure Supplementary 6, 7 and 8 - Feral Bees OOB plots << demography and selection >>
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS14_combined_plot_ne.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the population effective size $N_{\mathrm{e}}$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_NE}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS15_combined_plot_nc.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the population census size $N$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_N}
\end{figure}


\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS16_combined_plot_thetab.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the scaled mutation rate of beneficial mutations $\theta_{b}$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_thetab}
\end{figure}


% Figure Supplementary 17 to 21 - Feral Bees OOB plots << selection extra >>
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS17_combined_plot_prps.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the probability of beneficial mutation $P_RP_B$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_prpb}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS18_combined_plot_pstrong.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the proportion of strongly selected mutations $P$ for pairs of temporal samples of \textit{A. mellifera} feral populations.} Stacked points on the far left of each scatter-plot correspond to posterior estimates of neutral simulations, where $P = 0$. MSE and $R^2$ estimates corresponded to non-neutral PODs. See Table S2, Supplementary results, for MSE and $R^2$ estimates for neutral PODs.}
  \label{fig:supple_feralbee_pstrong}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS19_combined_plot_gamma.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the mean of the gamma distribution $\gamma$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_gammamean}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS20_combined_plot_gammasel.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the average selection coefficients of mutations under strong selection $\bar{s}$ for pairs of temporal samples of \textit{A. mellifera} feral populations.} Stacked points on the far left of each scatter-plot correspond to posterior estimates of neutral simulations, where $\bar{s} = 0$. MSE and $R^2$ estimates corresponded to non-neutral PODs. See Table S2, Supplementary results, for MSE and $R^2$ estimates for neutral PODs.}
  \label{fig:supple_feralbee_gammaselmean}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS21_combined_plot_load.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of the mean substitution load $L$ for pairs of temporal samples of \textit{A. mellifera} feral populations.} Stacked points on the far left of each scatter-plot correspond to posterior estimates of neutral simulations, where $L = 0$. MSE and $R^2$ estimates corresponded to non-neutral PODs. See Table S2, Supplementary results, for MSE and $R^2$ estimates for neutral PODs.}
  \label{fig:supple_feralbee_load}
\end{figure}

% Figure Supplementary 22 to 24 - Feral Bees OOB plots << demography extra >>
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS22_combined_plot_mu.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the mutation rate per generation $\mu$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_mu}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS23_combined_plot_rr.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the per base recombination rate per generation $r$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_r}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS24_combined_plot_nenc.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the the ratio between the effective size and the population census size for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_nen}
\end{figure}

% Figure Supplementary 25 NE Expected versus FST-NE prediction of Feral Bee populations.
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS25_combined_plot_fstNE.pdf}
  \small\caption{\textbf{$\hat N_{\mathrm{e}}$ estimates from the temporal $F_{\mathrm{ST}}$ to compare with the ABC-RF -based $\hat N_{\mathrm{e}}$ estimates for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_fstne}
\end{figure}

% Figure Supplementary 26 and 27 - Feral Bees Posterior density plots << selection and demography extra >>
\begin{figure}[ht]
  \centering 
  \includegraphics[width=0.95\textwidth]{Figures/FigureS26_weighted_densityPlots_selection_feralbees.pdf}
  \small\caption{\textbf{Posterior distributions of model parameters and a latent variable informative about selection for all feral \textit{A. mellifera} populations.} (a) probability of a beneficial mutation to arise $P_RP_B$; (b) number of mutations under strong selection $P$; (c)  mean of the gamma distribution $\gamma$; (d) mean of selection coefficients of mutations under strong selection $\bar{s}$; and (e) mean substitution load $L$. Dashed and filled lines corresponds to the prior and posterior distributions, respectively. See Table S3, Supplementary Results for mean estimates and 95\% credibility intervals.}
  \label{fig:supple_feralbee_densityselection}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS27_weighted_densityPlots_demography_feralbees.pdf}
  \small\caption{\textbf{Posterior distributions of model parameters and a latent variable informative about demography for all feral \textit{A. mellifera} populations.} (a) mutation rate per generation $\mu$; (b) per base recombination rate per generation $r$; and (c) the ratio between the effective size and the population census size $N_{\mathrm{e}}/N$. Dashed and filled lines corresponds to the prior and posterior distributions, respectively. See Table S3, Supplementary Results for mean estimates and 95\% credibility intervals.}
  \label{fig:supple_feralbee_densitydemo}
\end{figure}

\end{document}
