\documentclass[a4paper, 12pt]{article}


% basic package list
%\usepackage[T1]{fontenc}
\usepackage{fontspec}
\defaultfontfeatures{Mapping=tex-text}
\usepackage[margin=25mm]{geometry}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}

% other packages
\usepackage{xunicode}
\usepackage{xltxtra}
\usepackage{hyperref}         % hyperlinks
\usepackage{booktabs}         % professional-quality tables
\usepackage{indentfirst}      % to indent section first paragraph
\usepackage{url}              % simple URL typesetting
\usepackage{natbib}
\usepackage[modulo]{lineno}
\usepackage{sectsty}          % to change section font size
\usepackage[flushleft]{threeparttable} % table with note
\usepackage{mathtools}
\usepackage{indentfirst}
\usepackage{multirow}
\usepackage[labelfont=bf]{caption}
\usepackage{lineno}
\usepackage{courier}
\usepackage{rotating} % to rotate table vertically (for table S2)
\usepackage{longtable} % for tables longer than 1 page
%\usepackage{tablefootnote} % to add table footnote


% black hypelinks with no border
\hypersetup{
    colorlinks,
    citecolor=black,
    filecolor=black,
    linkcolor=black,
    urlcolor=black
}

% set additional parameters
\setcitestyle{authoryear,open={(},close={)}}
\graphicspath {{Figures/}}

\sectionfont{\fontsize{14}{19}\selectfont}
\subsectionfont{\fontsize{12}{17}\selectfont}

\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}

\setcounter{figure}{0}

% reduce page number font size
\renewcommand*{\thepage}{\footnotesize\arabic{page}}

\title{Supplementary Material for: Joint inference of adaptive and demographic history from temporal population genomic data}
\author{\small
            Vitor A. C. Pavinato$^{1,2,3}$, Stéphane De Mita$^4$, Jean-Michel Marin$^2$, Miguel de Navascués$^{1,5*}$}
\date{{\footnotesize %
    $^1$CBGP, INRAE, CIRAD, IRD, Montpellier SupAgro, Université de Montpellier, Montpellier, France\\%
    $^2$UMR Institut Montpelliérain Alexander Grothendieck, Université de Montpellier, France\\%  
    $^3$Entomology Dept., CFAES, The Ohio State University, Wooster, USA\\%
    $^4$UMR Interactions Arbres-Microorganismes, INRAE, France \\%
    $^5$Human Evolution, Department of Organismal Biology, Uppsala University, Uppsala, Sweden\\%  
    $^*$corresponding author: miguel.navascues@inrae.fr\\[2ex]%
    }
    \footnotesize\today    
}
\begin{document}
\maketitle




\newpage

% Supplementary Table 1 - Symbols

\section*{S1 Supplementary Methods}

\subsection*{S1.1 List of summary statistics}

Three groups of summary statistics were calculated for the reference table (\texttt{e.g.} training data): 1) summary statistics calculated SNP-by-SNPs (i.e., locus-specific summary statistics - LSS); 2) summary statistics calculated for nucleotide sequence (i.e., window summary statistics - WSS) - in this case, a window containing base positions was defined around each SNP in the dataset, and 3) summary statistics calculated globally by averaging the locus-specific or window-specific summary statistics (global summary statistics - GSS). Summary statistics about within-population diversity as $H_{\mathrm{E}}$ were calculated for each time point individually and marginally by pooling all individuals sampled together. Three window sizes were used for the summary statistics calculated in a window: 500bp, 5Kbp, and 10Kbp. We used the allele counts instead of frequency for the site-frequency spectrum, and all bins were included in the reference table.

\begin{enumerate}
 \item LSSs - Locus-specific summary statistics:
    \begin{enumerate}
     \item $D_{\mathrm{J}}$ - Jost's $D$ \citep{Jost:2008cs,Jost:2009hn};
        \item $F_{\mathrm{ST}}$ - Weir and Cockerham's $F_{\mathrm{ST}}$ \citep{Weir:1984dx};
        \item $H_{\mathrm{E}}$ - Expected Heterozygosity;
    \end{enumerate}
    \item WSSs - Window-specific summary statistics (window sizes - 500bp, 5Kbp and 10Kbp):
    \begin{enumerate}
        \item $D_{\mathrm{j}}$ - Jost's $D$;
        \item $F_{\mathrm{ST}}$ - Weir and Cockerham's $F_{\mathrm{ST}}$;
        \item $D_{\mathrm{a}}$ - Net distance between populations;
        \item $H_{\mathrm{E}}$ - Expected Heterozygosity;
        \item $S$ - Number of polymorphic sites;
        \item $\pi$ - Nucleotide diversity \citep{Nei:1979uk};
        \item $\theta_{\mathrm{W}}$ - Watterson's 4Nu estimator \citep{Watterson:1975bh};
        \item $D$ - Tajima's D \citep{Tajima:1989un};
    \end{enumerate}
    \item Global summary statistics:
    \begin{enumerate}
        \item All summary statistics enumerated above (including the within population calculations);
        \item SFS bins - binned Joint Site-frequency spectrum $SFS$ \citep{Ewing:2016gv};
        \item mean, variance, kurtosis, skewness, and the 5\% and 95\% quantiles of all intra-locus summary statistics presented above (LSSs and WSSs);
    \end{enumerate}
\end{enumerate}

\section*{S2 Supplementary Results}

\subsection*{S2.1 ABC-RF}

Figure~\ref{fig:supple_pods_priors} shows the prior distribution of the parameters and latent variables estimated with the ABC-RF. These distributions correspond to the values that were used to train ABC-Random forests.\\

Figure~\ref{fig:supple_oob_sel} shows the distribution of Random Forest OOB estimates on the observed values (the ones used to train the RF) for model parameters and other latent variables informative about selection. The ABC-RF regressions performed poorly on the inference of two parameters of the selection dynamics $P_{\mathrm{R}}P_{\mathrm{B}}$ and $\gamma$. On the other hand, the ABC-RF performed better on the latent variables $P$ and $\bar{s}$ (which can be viewed as the realization of the two parameters mentioned above). Each value of these latent variables was obtained from the simulation by 1) calculating the relative proportion of strongly beneficial mutations (those with $s > 1/N_{\mathrm{e}}$) over all possible segregating mutations, 2) averaging their $s$ values.\\

The estimation of substitution load $L$ also had a somehow good performance. It performed well for higher values but poorly for lower ones, as shown in the distribution of the estimated OOB values against the true values (Figure~\ref{fig:supple_oob_sel} e). Only strong adaptive dynamics (many loci under selection or strong selection) seem to leave informative polymorphism patterns about substitution load. The proportion of strongly beneficial mutations that were segregating in the population, $P$, can also be used as a complementary estimator to characterize the adaptive dynamics (it is complementary to $\theta_{\mathrm{b}}$ presented in the main text).\\

Figure~\ref{fig:supple_oob_demo} also presents the distribution OOB estimates with the true observed values for model parameters and the latent variables informative about drift. ABC-RF could not satisfactorily recover the recombination rate $r$ as expected for these model parameters, given the calculated summary statistics and the information contained in the temporal allele frequency dynamics. On the other hand, the ABC-RF estimated the mutation rate per nucleotide position per generation $\mu$ satisfactorily. The last latent variable also performed relatively well and similarly to the $L$. It could be used with other estimates to characterize the selection/drift dynamics when selection is more frequent.\\

To test the robustness of our ABC-RF-based framework, we applied the trained RFs on PODs with across-genome varying recombination rates. Figure~\ref{fig:supple_example_recomb} depicts how recombination rates were set on simulations. Simulations for ABC-RF training had fixed genome-wide recombination rates per generation $r$; Additional 1,000 simulations for testing the framework's robustness had varying recombination rates.
The scatterplots showed a similar dispersion pattern as the OOB plots when the true values were plotted against the estimated values (Figure~\ref{fig:supple_pods_variable_recomb}). Quantitatively, these PODs presented similar behavior regarding the MSE and $R^2$ estimates (Figure~\ref{fig:supple_pods_variable_recomb}, and Table~\ref{tab:supple_neutral_mse}).\\

Table~\ref{tab:supple_neutral_mse} contains MSE and bias for $P$, $\bar{s}$, and $L$ from neutral simulations. For training simulations (for model proof of concept and analysis of the feral bee populations), we first removed neutral simulations for these parameters (where values were zero) before RF training. Then each trained RF was used to make predictions on these neutral simulations. The MSE and bias correspond to the difference between the true values (zeros) and the predicted values in the original scale. For simulations used to test the robustness of our framework when there are varying rates of recombination, we calculated MSE and $R^2$ for non-neutral PODs (Figure~\ref{fig:supple_pods_variable_recomb}) and MSE and bias for neutral PODs.\\

Figure~\ref{fig:supple_pods_varplots_sel} and Figure~\ref{fig:supple_pods_varplots_demo} show the variable importance plots (VIP) of each ABC-RF that were grown to infer the parameters of our ABC-RF framework. In a nutshell, the VIP shows how many times the RF selected summary statistics during the training phase. Consequently, the VIP shows the relative importance of each summary statistic for training that parameter (more details in the main text). For the latent variables that account for the selection and drift dynamics, the top summary statistics were those calculated from the genome-wide distribution of locus-specific summary statistics that reflect their genome-wide distribution - variance, mean, skewness, kurtosis, and quantiles. RF frequently used summary statistics that inform about the distribution of locus-specific heterozygosity, calculated globally and calculated for the second sample. And also, summary statistics informative about the distribution of genome-wide intra-locus $F_{\mathrm{ST}}$, for the latent variables $P$ and $\bar{s}$. For $L$ summary statistics informative about the genetic divergence between samples: $F_{\mathrm{ST}}$, Da and Dj were more frequently selected.\\

\subsection*{S2.2 Analysis of temporal genomic data of feral populations of \textit{Apis mellifera}}

Before we trained the RF to analyze feral bee populations, we checked if our model could capture most of the features of the observed genomic data (target dataset). We run a principal component analysis (PCA) and projected pairs of principal components (PC) on a PCA plot for each population reference table containing only the summary statistics from the simulated data. We also ran a PCA on the observed (or target) dataset summary statistics and projected the same PCs on the PCA plot computed with simulated data. The model mostly capture features of the data, however some PCs showed the observed data point (in black) far from the simulated data cloud of points (in grey), indicating some model inadequacies possibly due to the fact that we did not include gene flow and/or admixture in our simulations (Figure~\ref{fig:supple_model_check_avalon}, Figure~\ref{fig:supple_model_check_humboldt}, Figure~\ref{fig:supple_model_check_davis}, Figure~\ref{fig:supple_model_check_stanislaus}, Figure~\ref{fig:supple_model_check_stebbins}, Figure~\ref{fig:supple_model_check_riverside}, Figure~\ref{fig:supple_model_check_placerita}).\\

Figure~\ref{fig:supple_feralbee_thetab}, Figure~\ref{fig:supple_feralbee_N}, and Figure~\ref{fig:supple_feralbee_NE} show the distribution of the true and the OOB estimated values for the parameters used in this work for the joint inference of demography and selection for temporal pairs of populations of feral \textit{A. mellifera}. The OOB MSE and $R^2$ estimates showed that even with fewer simulations for model training, the RF grew for the \textit{A. mellifera} data had comparable performance to the RF grew for the simulated data.\\

The performance of the model parameters and additional latent variables for all temporal pairs of this dataset were similar to the simulations used for the framework evaluation. Figure~\ref{fig:supple_feralbee_prpb}, Figure~\ref{fig:supple_feralbee_pstrong}, Figure~\ref{fig:supple_feralbee_gammamean}, Figure~\ref{fig:supple_feralbee_gammaselmean}, and Figure~\ref{fig:supple_feralbee_load} show the parameters and latent variables informative about the adaptive history. Figure~\ref{fig:supple_feralbee_mu}, Figure~\ref{fig:supple_feralbee_r}, and Figure~\ref{fig:supple_feralbee_nen} show the model parameters and an additional latent variable complementary to the mutation load for the inference of the selection impact on demography. For comparison purposes, we also included the scatterplots of true $N_{\mathrm{e}}$ against the $F_{\mathrm{ST}}$-NE computed on the global $F_{\mathrm{ST}}$ of simulated data for all feral bee populations (Figure~\ref{fig:supple_feralbee_fstne}).\\

%Table~\ref{tab:supple_bees_posteriors}
Table S3 contains the posterior estimates of each parameter for each \textit{A. mellifera} population. The last row shows the $F_{\mathrm{ST}}$-NE estimates for comparison. Figure~\ref{fig:supple_feralbee_densityselection} and Figure~\ref{fig:supple_feralbee_densitydemo} show the prior, and posterior distributions of all remaining model parameters and latent variables calculated for the temporal samples analyzed with the ABC-RF framework.\\





\bibliographystyle{apalike}
\bibliography{references}
\newpage

\begin{table}[ht]
 \caption{\textbf{Notation.}}
  %\centering
  \label{table:tableS1}
  \begin{tabular}{ll}
   \cmidrule(r){1-2}
    Symbol                  & Meaning \\
    \midrule
    $r$        & recombination rate per base pair per generation \\
    $G$                     & genome size\\
    $h$                     & dominance coefficient\\
    $L$                     & mean substitution load\\
    $N$                     & population size during the inference period\\ 
    $N_{\mathrm{0}}$        & population size during the burn-in period\\
    $N_{\mathrm{e}}$        & effective population size during the inference period\\
    $P_{\mathrm{R}}$        & proportion of non-neutral regions\\
    $P_{\mathrm{B}}$        & proportion of beneficial mutations per non-neutral region\\
    $P$                     & proportion of polymorphisms under strong selection, i.e. $s > 1/ N_{\mathrm{e}}$\\
    $s$                     & selection coefficient\\
    $\bar{s}$               & average selection coefficients of mutations under strong selection\\
    $t_1$ and $t_2$         & generations were samples of individuals were taken\\
    $W_{\mathrm{max}i}$     & fitness of individual with highest fitness in the population in generation $i$\\
    $\bar{W_{i}}$           & mean fitness of the population in generation $i$\\
    $\gamma$                & mean of the gamma distribution where the selection coefficients of each\\
                            & beneficial mutation was sampled\\
    $\mu$                   & mutation rate per generation\\
    $\mu_{\mathrm{b}}$      & mutation rate of beneficial mutations per generation\\
    $\tau$                  & length of time-interval between the first and the second sample\\
    $\theta_{\mathrm{b}}$   & scaled mutation rate of beneficial mutations\\
&\\
    
    \bottomrule
  \end{tabular}
  \label{tab:supple_symbols}
\end{table}




% Supplementary Table 2 - MSE and BIAS for neutral/near-neutral simulations
% Please add the following required packages to your document preamble:
% \usepackage{multirow}
\begin{table}[ht]
\caption{\textbf{Neutral simulations' MSE and Bias.}}
\centering
%\label{table:tableS2}
\begin{tabular}{clll}
\hline
\multicolumn{1}{l}{Parameter}            & Dataset            & MSE   & Bias  \\ 
\hline
\multirow{9}{*}{$P$}                            & Training           & 0.001  & 0.014  \\
                                                           & PODs recombination & 0.001 & 0.013 \\
                                                           & Avalon             &  0     & 0.007 \\
                                                           & Humboldt        & 0      & 0.009  \\
                                                           & Davis               & 0      & 0.009  \\
                                                           & Stanislaus       & 0       & 0.010 \\
                                                           & Stebbins          & 0      & 0.013  \\
                                                           & Riverside         & 0      & 0.013  \\
                                                           & Placerita          &  0     & 0.013  \\ 
\hline
\multirow{9}{*}{$\bar{s}$}                    & Training           & 0.262  & 0.401 \\
                                                           & PODs recombination & 0.241 & 0.386 \\
                                                           & Avalon             & 0.238           & 0.410 \\
                                                           & Humboldt           & 0.318 &0.471 \\
                                                           & Davis              & 0.305 & 0.467 \\
                                                           & Stanislaus         & 0.315 & 0.479 \\
                                                           & Stebbins           & 0.373 & 0.533\\
                                                           & Riverside          & 0.368 & 0.540\\
                                                           & Placerita          & 0.379 & 0.542\\ 
\hline
\multirow{9}{*}{$L$}                            & Training           & 0.014  & 0.089 \\
                                                           & PODs recombination & 0.013 & 0.088 \\
                                                           & Avalon             & 0.011 & 0.031 \\
                                                           & Humboldt        & 0.006 & 0.024\\
                                                           & Davis              & 0.010 & 0.034\\
                                                           & Stanislaus       & 0.007 & 0.032\\
                                                           & Stebbins          & 0.008 & 0.033 \\
                                                           & Riverside         & 0.011 & 0.046\\
                                                           & Placerita          & 0.008 & 0.034\\ 
\hline
\end{tabular}
\label{tab:supple_neutral_mse}
\end{table}

\newpage

% Supplementary Table  3 - Posterior estimates parameter/population.
 \begin{sidewaystable}[!ht]
 \begin{small}
 \caption{\textbf{\small Parameter posterior median and 95\% credibility interval (in parenthesis) for each  \textit{A. mellifera} feral population.}}
 \end{small}
   \begin{tiny}
    %\centering
    \begin{tabular}{cccccccc}
    \cmidrule(r){1-8}
        Parameter & Avalon & Humboldt & Davis & Stanislaus & Stebbins & Riverside & Placerita \\
        \midrule
        $P_\mathrm{R}P_\mathrm{B}$  & 1.53E-04 & 8.35E-04 & 1.26E-04 & 2.39E-04 & 9.70E-05 & 8.24E-05 & 6.19E-04 \\
         & (1.59E-07 – 2.48E-01) & (1.55E-06 – 2.02E-01) & (1.17E-07 – 4.99E-02) & (4.32E-06 – 1.70E-01) & (1.14E-07 – 4.64E-02) & (1.18E-07 – 5.45E-01) & (4.15E-07 – 3.54E-01) \\
        $P$ & 1.48E-05 & 4.12E-05 & 1.89E-05 & 3.26E-05 & 2.61E-05 & 3.26E-05 & 9.39E-05 \\
         & (1.45E-06 – 1.04E-03) & (8.58E-06 – 3.08E-03) & (1.82E-06 – 3.99E-04) & (2.16E-06 – 3.49E-04) & (2.95E-06 – 5.23E-04) & (3.32E-06 – 4.93E-01) & (4.90E-06 – 4.93E-01) \\
        $\gamma$ & 1.77E-02 & 2.67E-02 & 2.14E-02 & 2.29E-02 & 2.35E-02 & 3.20E-02 & 2.67E-02 \\
         & (1.12E-03 – 8.82E-01) & (1.22E-03 – 8.45E-01) & (1.12E-03 – 9.12E-01) & (1.05E-03 – 8.93E-01) & (1.04E-03 – 8.93E-01) & (1.03E-03 – 8.93E-01) & (1.15E-03 – 8.92E-01) \\
        $\bar{s}$ & 9.72E-02 & 3.07E-01 & 1.75E-01 & 2.30E-01 & 1.58E-01 & 2.19E-01 & 3.31E-01 \\
         & (4.31E-03 – 1.62E+00) & (3.22E-02 – 2.23E+00) & (1.95E-03 – 1.50E+00) & (1.95E-03 – 1.30E+00) & (1.31E-03 – 1.51E+00) & (1.46E-03 – 2.25E+00) & (6.17E-03 – 1.98E+00) \\
        $\theta_{\mathrm{b}}$ & 3.72E-01 & 1.73E+00 & 2.90E+00 & 3.46E+00 & 3.07E+00 & 2.90E+00 & 6.22E+00 \\
         & (1.72E-04 – 5.66E+03) & (1.22E-03 – 1.56E+04) & (1.70E-02 – 1.32E+03) & (2.66E-02 – 1.37E+03) & (9.20E-03 – 4.61E+02) & (4.41E-03 – 2.04E+04) & (1.30E-02 – 1.70E+04) \\
        $L$ & 1.01E-03 & 1.59E-04 & 2.06E-01 & 2.65E-01 & 2.16E-04 & 9.43E-03 & 2.59E-02 \\
         & (1.14E-09 – 9.64E-01) & (1.42E-16 – 9.27E-01) & (5.32E-08 – 8.29E-01) & (1.22E-06 – 8.78E-01) & (2.00E-16 – 4.62E-01) & (1.47E-08 – 8.71E-01) & (2.01E-16 – 7.56E-01) \\
        $\mu$ & 5.67E-08 & 8.05E-08 & 5.03E-08 & 2.56E-08 & 7.12E-08 & 2.98E-08 & 1.11E-07 \\
         & (3.45E-09 – 1.65E-07) & (9.75E-09 – 1.66E-07) & (4.04E-09 – 1.57E-07) & (3.06E-09 – 1.45E-07) & (4.86E-09 – 1.66E-07) & (2.71E-09 – 1.65E-07) & (1.23E-08 – 1.67E-07) \\
        $r$ & 3.72E-08 & 5.19E-08 & 2.71E-08 & 3.19E-08 & 2.28E-08 & 2.10E-08 & 3.54E-08 \\
         & (1.02E-08 – 6.14E-06) & (1.02E-08 – 2.06E-06) & (1.02E-08 – 2.01E-06) & (1.04E-08 – 1.06E-06) & (1.02E-08 – 8.54E-07) & (1.02E-08 – 1.06E-06) & (1.02E-08 – 3.64E-06) \\
        $N$ & 8.10E+01 & 4.90E+01 & 1.02E+03 & 1.80E+03 & 3.42E+02 & 5.19E+02 & 1.39E+02 \\
         & (1.70E+01 – 5.36E+03) & (1.00E+01 – 7.71E+03) & (2.42E+02 – 6.78E+03) & (3.67E+02 – 8.00E+03) & (1.58E+02 – 3.09E+03) & (6.30E+01 – 8.72E+03) & (1.40E+01 – 4.69E+03) \\
        $N_{\mathrm{e}}$ & 8.30E+01 & 4.42E+01 & 1.04E+03 & 1.96E+03 & 3.80E+02 & 5.69E+02 & 1.46E+02 \\
         & (2.21E+01 – 9.59E+02) & (4.55E+00 – 5.76E+02) & (2.03E+02 – 7.24E+03) & (4.19E+02 – 8.70E+03) & (1.76E+02 – 3.90E+03) & (5.53E+01 – 8.17E+03) & (4.57E+00 – 1.15E+03) \\
        $N_{\mathrm{e}}/N$ & 9.07E-01 & 1.05E+00 & 1.07E+00 & 1.06E+00 & 1.09E+00 & 1.04E+00 & 1.06E+00 \\
         & (1.15E-03 – 1.13E+00) & (8.75E-04 – 1.17E+00) & (5.63E-01 – 1.12E+00) & (5.26E-01 – 1.12E+00) & (2.28E-01 – 1.13E+00) & (2.78E-01 – 1.13E+00) & (2.44E-01 – 1.14E+00) \\
         \midrule
        $F_{\mathrm{ST}}$-$N_{\mathrm{e}}$ & 8.00E+01 & 1.13E+02 & 3.62E+01 & 2.28E+01 & 3.60E+01 & 1.47E+01 & 6.15E+01 \\
         & (0.00E+00 – 7.80E+02) & (5.30E+00 – 5.88E+17) & (8.63E+00 – 3.59E+02) & (6.00E+00 – 3.42E+02) & (5.63E+00 – 2.25E+02) & (1.79E+00 – 1.20E+02) & (4.46E+00 – 6.12E+02) \\
          \bottomrule
    \multicolumn{8}{l}{\tiny  Estimates are in the original scale.} \\
    \end{tabular}
\label{tab:supple_bees_posteriors}
\end{tiny}
\end{sidewaystable}

\newpage


\begin{tiny}
\begin{longtable}{p{3.0cm}p{1.0cm}p{1.0cm}p{1.0cm}p{1.0cm}p{1.0cm}p{1.0cm}p{1.0cm}}
\caption{\textbf{\small Summary statistics for \textit{A. mellifera} feral populations.}}\label{tab:sumstats_bee} \\

\hline & \multicolumn{1}{l}{Avalon} & \multicolumn{1}{l}{Humboldt} & \multicolumn{1}{l}{Davis} & \multicolumn{1}{l}{Stanislaus} & \multicolumn{1}{l}{Stebbins} & \multicolumn{1}{l}{Riverside} & \multicolumn{1}{l}{Placerita} \\ \hline 
\endfirsthead

\multicolumn{8}{r}{\textit{continued from previous page}} \\
\hline \multicolumn{1}{r}{} & \multicolumn{1}{l}{Avalon} & \multicolumn{1}{l}{Humboldt} & \multicolumn{1}{l}{Davis} & \multicolumn{1}{l}{Stanislaus} & \multicolumn{1}{l}{Stebbins} & \multicolumn{1}{l}{Riverside} & \multicolumn{1}{l}{Placerita} \\ \hline 
\endhead

\hline \multicolumn{8}{r}{\textit{continued on next page}} \\
\endfoot

\hline
\endlastfoot

GSS $H_{\mathrm{E}}$ & 0.37 & 0.29 & 0.28 & 0.28 & 0.25 & 0.22 & 0.21 \\
GSS $D_{\mathrm{J}}$ & 0.17 & 0.12 & 0 & 0 & 0 & 0 & 0.02 \\
GSS $F_{\mathrm{ST}}$ & 0.28 & 0.26 & 0.02 & 0.01 & 0.01 & 0.01 & 0.05 \\
GSS $S$ & 43187 & 61431 & 51268 & 61264 & 69961 & 113897 & 98438 \\
GSS $\theta_{\mathrm{W}}$ & 13916.88 & 16450.51 & 16121.33 & 18462.86 & 19719.88 & 32104.1 & 27746.68 \\
GSS $\pi$ & 16106.09 & 17784.77 & 14465.06 & 16938.75 & 17257.82 & 24881.59 & 20222.57 \\
GSS $D$ & 0.73 & 0.33 & -0.47 & -0.36 & -0.52 & -0.94 & -1.14 \\
GSS $D_{\mathrm{a}}$ & 0.14 & 0.08 & 0 & 0 & 0 & 0 & 0.01 \\
GSS $t_1$ $H_{\mathrm{E}}$ & 0.15 & 0.26 & 0.29 & 0.29 & 0.26 & 0.2 & 0.15 \\
GSS $t_2$ $H_{\mathrm{E}}$ & 0.36 & 0.23 & 0.28 & 0.27 & 0.23 & 0.22 & 0.23 \\
GSS $t_1$ $S$ & 8280 & 46969 & 27675 & 32530 & 54885 & 43351 & 40960 \\
GSS $t_2$ $S$ & 36347 & 45561 & 43311 & 54571 & 48762 & 103077 & 83254 \\
GSS $t_1$ $\theta{\mathrm{W}}$ & 5520 & 15553.28 & 15095.45 & 17743.64 & 19401.07 & 23646 & 15797.25 \\
GSS $t_2$ $\theta{\mathrm{W}}$ & 12848.15 & 15087.04 & 15309.82 & 18070.6 & 17236.67 & 31063.86 & 27568.67 \\
GSS $t_1$ $\pi$  & 5592 & 16163.6 & 15055.33 & 17661 & 18281.3 & 23115.33 & 14802.06 \\
GSS $t_2$ $\pi$  & 15535.81 & 14006.38 & 14116 & 16630.37 & 15978.22 & 25182.07 & 22708.71 \\
GSS $t_1$ $D$ & 0 & 0.19 & -0.03 & -0.05 & -0.29 & -0.24 & -0.35 \\
GSS $t_2$ $D$  & 1.05 & -0.34 & -0.39 & -0.38 & -0.37 & -0.83 & -0.83 \\
SFS bin$_1$ & 0 & 16110 & 0 & 0 & 25318 & 48723 & 39536 \\
SFS bin$_2$ & 9364 & 7870 & 20589 & 22294 & 12257 & 23856 & 24263 \\
SFS bin$_3$ & 3403 & 5514 & 11095 & 11758 & 8423 & 12827 & 11460 \\
SFS bin$_4$ & 4781 & 4394 & 182 & 8575 & 6313 & 8346 & 6628 \\
SFS bin$_5$ & 5623 & 6770 & 7313 & 6212 & 4818 & 5783 & 4684 \\
SFS bin$_6$ & 4580 & 2931 & 4588 & 6 & 3852 & 4285 & 3159 \\
SFS bin$_7$ & 3895 & 3319 & 72 & 4694 & 3062 & 3419 & 2852 \\
SFS bin$_8$ & 4016 & 3801 & 3380 & 3428 & 2539 & 2997 & 2331 \\
SFS bin$_9$ & 4572 & 4094 & 2752 & 2901 & 2280 & 2509 & 2159 \\
SFS bin$_10$ & 2953 & 6628 & 1297 & 1396 & 1099 & 1152 & 1366 \\
LSS mean($H_{\mathrm{E}}$) & 0.37 & 0.29 & 0.28 & 0.28 & 0.25 & 0.22 & 0.21 \\
LSS mean($D_{\mathrm{J}}$) & 0.17 & 0.12 & 0 & 0 & 0 & 0 & 0.02 \\
LSS mean($F_{\mathrm{ST}}$) & 0.15 & 0.14 & -0.01 & -0.02 & 0.01 & -0.02 & 0.02 \\
LSS $t_1$ mean($H_{\mathrm{E}}$) & 0.15 & 0.26 & 0.29 & 0.29 & 0.26 & 0.2 & 0.15 \\
LSS $t_2$ mean($H_{\mathrm{E}}$) & 0.36 & 0.23 & 0.28 & 0.27 & 0.23 & 0.22 & 0.23 \\
WSS mean($H_{\mathrm{E}}$) & 0.37 & 0.29 & 0.28 & 0.28 & 0.25 & 0.22 & 0.21 \\
WSS mean($D_{\mathrm{J}}$) & 0.17 & 0.12 & 0 & 0 & 0 & 0 & 0.02 \\
WSS mean($F_{\mathrm{ST}}$) & 0.23 & 0.22 & 0.01 & 0 & 0.01 & 0 & 0.05 \\
WSS mean($S$) & 7.07 & 9.3 & 8.02 & 9.28 & 10.39 & 16.15 & 14.17 \\
WSS mean($\theta_{\mathrm{W}}$) & 2.3 & 2.49 & 2.52 & 2.8 & 2.93 & 4.55 & 4 \\
WSS mean($\pi$) & 2.64 & 2.7 & 2.28 & 2.57 & 2.57 & 3.53 & 2.91 \\
WSS mean($D$) & 0.53 & 0.26 & -0.37 & -0.29 & -0.42 & -0.82 & -0.97 \\
WSS mean($D_{\mathrm{a}}$) & 0.14 & 0.08 & 0 & 0 & 0 & 0 & 0.01 \\
WSS $t_1$ mean($H_{\mathrm{E}}$) & 0.15 & 0.26 & 0.29 & 0.29 & 0.26 & 0.2 & 0.15 \\
WSS $t_2$ mean($H_{\mathrm{E}}$) & 0.36 & 0.23 & 0.28 & 0.27 & 0.23 & 0.22 & 0.23 \\
WSS $t_1$ mean($S$) & 1.3 & 7.13 & 4.35 & 4.92 & 8.11 & 6.14 & 5.81 \\
WSS $t_2$ mean($S$) & 6.02 & 6.93 & 6.81 & 8.29 & 7.3 & 14.61 & 12.07 \\
WSS $t_1$ mean($\theta_{\mathrm{W}}$) & 0.86 & 2.36 & 2.38 & 2.69 & 2.87 & 3.35 & 2.25 \\
WSS $t_2$ mean($\theta_{\mathrm{W}}$) & 2.13 & 2.29 & 2.41 & 2.74 & 2.58 & 4.4 & 4 \\
WSS $t_1$ mean($\pi$) & 0.88 & 2.45 & 2.37 & 2.67 & 2.71 & 3.28 & 2.1 \\
WSS $t_2$ mean($\pi$) & 2.57 & 2.13 & 2.22 & 2.52 & 2.38 & 3.57 & 3.28 \\
WSS $t_1$ mean($D$) & 0.78 & 0.14 & -0.06 & -0.05 & -0.24 & -0.21 & -0.32 \\
WSS $t_2$ mean($D$) & 0.83 & -0.28 & -0.32 & -0.31 & -0.31 & -0.72 & -0.72 \\
LSS variance($H_{\mathrm{E}}$) & 0.02 & 0.03 & 0.02 & 0.02 & 0.02 & 0.02 & 0.02 \\
LSS variance($D_{\mathrm{J}}$) & 0.08 & 0.04 & 0 & 0.02 & 0 & 0.01 & 0.01 \\
LSS variance($F_{\mathrm{ST}}$) & 0.12 & 0.05 & 0.04 & 0.04 & 0.01 & 0.03 & 0.01 \\
LSS $t_1$ variance($H_{\mathrm{E}}$) & 0.09 & 0.04 & 0.08 & 0.08 & 0.03 & 0.07 & 0.04 \\
LSS $t_2$ variance($H_{\mathrm{E}}$) & 0.04 & 0.03 & 0.03 & 0.03 & 0.03 & 0.02 & 0.02 \\
WSS variance($H_{\mathrm{E}}$) & 0.01 & 0.01 & 0 & 0 & 0 & 0 & 0 \\
WSS variance($D_{\mathrm{J}}$) & 0.03 & 0.01 & 0 & 0 & 0 & 0 & 0 \\
WSS variance($F_{\mathrm{ST}}$) & 0.05 & 0.02 & 0.02 & 0.02 & 0.01 & 0.01 & 0.01 \\
WSS variance($S$) & 18.52 & 27.66 & 21 & 27.23 & 33.73 & 71.26 & 63.45 \\
WSS variance($\theta_{\mathrm{W}}$) & 1.98 & 1.98 & 2.08 & 2.47 & 2.68 & 5.66 & 5.07 \\
WSS variance($\pi$) & 3.06 & 3.01 & 2.01 & 2.5 & 2.43 & 4.18 & 3.07 \\
WSS variance($D$) & 0.75 & 0.69 & 0.56 & 0.55 & 0.5 & 0.32 & 0.33 \\
WSS variance($D_{\mathrm{a}}$) & 0.02 & 0.01 & 0 & 0 & 0 & 0 & 0 \\
WSS $t_1$ variance($H_{\mathrm{E}}$) & 0.03 & 0.01 & 0.02 & 0.02 & 0.01 & 0.01 & 0.01 \\
WSS $t_2$ variance($H_{\mathrm{E}}$) & 0.01 & 0.01 & 0.01 & 0.01 & 0.01 & 0 & 0 \\
WSS $t_1$ variance($S$) & 1.74 & 20.22 & 10.58 & 12.44 & 22.62 & 18.31 & 15.17 \\
WSS $t_2$ variance($S$) & 15.87 & 20.1 & 17.83 & 23.65 & 21.43 & 60.91 & 51.58 \\
WSS $t_1$ variance($\theta_{\mathrm{W}}$) & 0.77 & 2.22 & 3.19 & 3.7 & 2.82 & 5.45 & 2.29 \\
WSS $t_2$ variance($\theta_{\mathrm{W}}$) & 1.98 & 2.21 & 2.23 & 2.59 & 2.68 & 5.53 & 5.66 \\
WSS $t_1$ variance($\pi$) & 0.85 & 2.61 & 3.21 & 3.7 & 2.92 & 5.26 & 2.17 \\
WSS $t_2$ variance($\pi$) & 3.23 & 2.21 & 2.09 & 2.48 & 2.38 & 4.31 & 4.04 \\
WSS $t_1$ variance($D$) & 1.95 & 0.65 & 0.61 & 0.58 & 0.52 & 0.38 & 0.55 \\
WSS $t_2$ variance($D$) & 0.87 & 0.65 & 0.61 & 0.59 & 0.6 & 0.33 & 0.37 \\
LSS kurtosis($H_{\mathrm{E}}$) & 1.74 & 1.42 & 1.83 & 1.74 & 1.89 & 2.54 & 2.9 \\
LSS kurtosis($D_{\mathrm{J}}$) & 5.1 & 4.6 & 0 & 12.05 & 18.94 & 18.45 & 25.7 \\
LSS kurtosis($F_{\mathrm{ST}}$) & 3.04 & 3.47 & 4.88 & 5.17 & 7.23 & 5.61 & 8.51 \\
LSS $t_1$ kurtosis($H_{\mathrm{E}}$) & 4.44 & 1.62 & 1.32 & 1.18 & 1.9 & 1.47 & 2.39 \\
LSS $t_1$ kurtosis($H_{\mathrm{E}}$) & 2.19 & 1.86 & 2.1 & 2.04 & 1.82 & 2.42 & 2.39 \\
WSS kurtosis($H_{\mathrm{E}}$) & 2.98 & 3.35 & 3.42 & 3.43 & 3.83 & 5.5 & 5.32 \\
WSS kurtosis($D_{\mathrm{J}}$) & 6.33 & 6.02 & 0 & 16.68 & 15.31 & 13.31 & 37.84 \\
WSS kurtosis($F_{\mathrm{ST}}$) & 3.4 & 2.84 & 6.09 & 5.97 & 6.58 & 5.67 & 6.01 \\
WSS kurtosis($\pi$) & 18.42 & 14.94 & 7.94 & 13.06 & 11.08 & 13.93 & 14.56 \\
WSS kurtosis($D$) & 2.68 & 2.69 & 2.85 & 2.8 & 3 & 3.71 & 3.84 \\
WSS kurtosis($D_{\mathrm{a}}$) & 9.82 & 7.77 & 41.89 & 37.07 & 19.05 & 15.6 & 56.6 \\
WSS $t_1$ kurtosis($H_{\mathrm{E}}$) & 6.45 & 3.34 & 2.68 & 2.8 & 3.7 & 3.94 & 4.49 \\
WSS $t_2$ kurtosis($H_{\mathrm{E}}$) & 3.57 & 3.41 & 3.66 & 3.67 & 3.67 & 5.31 & 5.16 \\
WSS $t_1$ kurtosis($S$) & 5.61 & 12.23 & 8.21 & 8.58 & 9.46 & 13.76 & 10.17 \\
WSS $t_2$ kurtosis($S$) & 13.32 & 13.22 & 10.47 & 9.74 & 12.76 & 8.21 & 17.75 \\
WSS $t_1$ kurtosis($\theta_{\mathrm{W}}$) & 6.46 & 12.23 & 8.66 & 8.58 & 9.46 & 13.76 & 10.35 \\
WSS $t_2$ kurtosis($\theta_{\mathrm{W}}$) & 13.32 & 13.25 & 10.46 & 9.74 & 12.76 & 8.21 & 17.73 \\
WSS $t_1$ kurtosis($\pi$) & 6.52 & 10.15 & 8.33 & 8.19 & 11.37 & 13.94 & 9.96 \\
WSS $t_2$ kurtosis($\pi$) & 18.8 & 12.82 & 8.08 & 9.59 & 8.21 & 12.68 & 15.89 \\
WSS $t_1$ kurtosis($D$) & 5.29 & 2.56 & 4.82 & 3.19 & 2.77 & 4.51 & 2.93 \\
WSS $t_2$ kurtosis($D$) & 2.92 & 2.62 & 2.64 & 2.74 & 2.67 & 3.58 & 3.52 \\
LSS skewness($H_{\mathrm{E}}$) & -0.39 & 0.08 & 0.47 & 0.39 & 0.54 & 0.89 & 1.08 \\
LSS skewness($D_{\mathrm{J}}$) & 1.22 & 1.58 & 0 & 2.09 & 2.95 & 2.59 & 3.8 \\
LSS skewness($F_{\mathrm{ST}}$) & 0.78 & 1.14 & 1.39 & 1.58 & 1.65 & 1.81 & 1.88 \\
LSS $t1$ skewness($H_{\mathrm{E}}$) & 1.68 & -0.03 & 0.02 & 0 & 0.01 & 0.58 & 0.88 \\
LSS $t2$ skewness($H_{\mathrm{E}}$) & -0.77 & 0.25 & -0.03 & 0.1 & 0.2 & 0.58 & 0.39 \\
WSS skewness($H_{\mathrm{E}}$) & -0.42 & 0.08 & 0.36 & 0.32 & 0.4 & 0.72 & 0.77 \\
WSS skewness($D_{\mathrm{J}}$) & 1.25 & 1.36 & 0 & 2.08 & 2.18 & 1.51 & 3.19 \\
WSS skewness($F_{\mathrm{ST}}$) & 0.51 & 0.44 & 1.37 & 1.4 & 1.47 & 1.16 & 1.21 \\
WSS skewness($\pi$) & 2.48 & 2.28 & 1.57 & 1.98 & 1.83 & 2.11 & 2.18 \\
WSS skewness($D$) & -0.4 & 0.01 & 0.32 & 0.26 & 0.36 & 0.57 & 0.69 \\
WSS skewness($D_{\mathrm{a}}$) & 1.99 & 1.69 & 3.5 & 3.43 & 2.62 & 1.93 & 3.89 \\
WSS $t_1$ skewness($H_{\mathrm{E}}$) & 1.52 & 0 & -0.07 & -0.04 & 0.01 & 0.37 & 0.61 \\
WSS $t_2$ skewness($H_{\mathrm{E}}$) & -0.64 & 0.12 & -0.15 & 0.06 & 0.07 & 0.5 & 0.21 \\
WSS $t_1$ skewness($S$) & 1.34 & 1.96 & 1.58 & 1.64 & 1.63 & 2.06 & 1.79 \\
WSS $t_2$ skewness($S$) & 2.05 & 2.06 & 1.74 & 1.64 & 1.91 & 1.5 & 2.42 \\
WSS $t_1$ skewness($\theta_{\mathrm{W}}$) & 1.41 & 1.96 & 1.65 & 1.64 & 1.63 & 2.06 & 1.8 \\
WSS $t_2$ skewness($\theta_{\mathrm{W}}$) & 2.05 & 2.07 & 1.74 & 1.64 & 1.91 & 1.5 & 2.42 \\
WSS $t_1$ skewness($\pi$) & 1.48 & 1.8 & 1.62 & 1.61 & 1.91 & 2.08 & 1.79 \\
WSS $t_2$ skewness($\pi$) & 2.49 & 2.12 & 1.54 & 1.69 & 1.52 & 2.01 & 2.3 \\
WSS $t_1$ skewness($D$) & 1.2 & -0.04 & 1.24 & 1 & 0.26 & 1.29 & 0.36 \\
WSS $t_2$ skewness($D$) & -0.68 & 0.28 & 0.3 & 0.3 & 0.32 & 0.53 & 0.61 \\
LSS quantile(0.05,$H_{\mathrm{E}}$) & 0.14 & 0.08 & 0.14 & 0.13 & 0.1 & 0.1 & 0.09 \\
LSS quantile(0.05, $D_{\mathrm{J}}$) & -0.13 & -0.04 & -0.16 & -0.15 & -0.09 & -0.13 & -0.06 \\
LSS quantile(0.05, $F_{\mathrm{ST}}$) & -0.32 & -0.09 & -0.2 & -0.2 & -0.11 & -0.17 & -0.11 \\
LSS $t_1$ quantile(0.05,$H_{\mathrm{E}}$) & 0 & 0 & 0 & 0 & 0 & 0 & 0 \\
LSS $t_2$ quantile(0.05,$H_{\mathrm{E}}$) & 0 & 0 & 0 & 0 & 0 & 0 & 0 \\
WSS quantile(0.05,$H_{\mathrm{E}}$) & 0.22 & 0.17 & 0.17 & 0.17 & 0.15 & 0.15 & 0.13 \\
WSS quantile(0.05, $D_{\mathrm{J}}$) & -0.03 & 0 & -0.09 & -0.08 & -0.04 & -0.04 & -0.02 \\
WSS quantile(0.05, $F_{\mathrm{ST}}$) & -0.1 & -0.01 & -0.15 & -0.15 & -0.07 & -0.11 & -0.05 \\
WSS quantile(0.05, $S$) & 2 & 3 & 2 & 3 & 3 & 5 & 4 \\
WSS quantile(0.05, $\theta_{\mathrm{W}}$) & 0.64 & 0.8 & 0.63 & 0.9 & 0.85 & 1.41 & 1.13 \\
WSS quantile(0.05, $\pi$) & 0.55 & 0.6 & 0.53 & 0.63 & 0.59 & 0.96 & 0.74 \\
WSS quantile(0.05, $D$) & -1.1 & -1.15 & -1.51 & -1.48 & -1.51 & -1.66 & -1.8 \\
WSS quantile(0.05, $D_{\mathrm{a}}$) & 0 & 0 & -0.05 & -0.05 & -0.02 & -0.03 & -0.01 \\
WSS $t_1$ quantile(0.05,$H_{\mathrm{E}}$) & 0 & 0.12 & 0 & 0.06 & 0.13 & 0.06 & 0.04 \\
WSS $t_2$ quantile(0.05,$H_{\mathrm{E}}$) & 0.18 & 0.08 & 0.13 & 0.15 & 0.09 & 0.14 & 0.15 \\
WSS $t_1$ quantile(0.05, $S$) & 0 & 2 & 0 & 1 & 2 & 1 & 1 \\
WSS $t_2$ quantile(0.05, $S$) & 1 & 1 & 1 & 2 & 1 & 4 & 3 \\
WSS $t_1$ quantile(0.05, $\theta_{\mathrm{W}}$) & 0 & 0.66 & 0 & 0.55 & 0.71 & 0.55 & 0.39 \\
WSS $t_2$ quantile(0.05, $\theta_{\mathrm{W}}$) & 0.35 & 0.33 & 0.35 & 0.66 & 0.35 & 1.21 & 0.99 \\
WSS $t_1$ quantile(0.05, $\pi$) & 0 & 0.48 & 0 & 0.5 & 0.57 & 0.5 & 0.25 \\
WSS $t_2$ quantile(0.05, $\pi$) & 0.47 & 0.33 & 0.4 & 0.55 & 0.4 & 0.95 & 0.8 \\
WSS $t_1$ quantile(0.05, $D$) & -0.71 & -1.18 & -0.81 & -0.81 & -1.4 & -0.82 & -1.45 \\
WSS $t_2$ quantile(0.05, $D$) & -1.06 & -1.53 & -1.56 & -1.53 & -1.49 & -1.59 & -1.63 \\
LSS quantile(0.95,$H_{\mathrm{E}}$) & 0.55 & 0.52 & 0.53 & 0.53 & 0.51 & 0.51 & 0.51 \\
LSS quantile(0.95, $D_{\mathrm{J}}$) & 0.89 & 0.59 & 0.28 & 0.18 & 0.09 & 0.13 & 0.15 \\
LSS quantile(0.95, $F_{\mathrm{ST}}$) & 0.84 & 0.59 & 0.3 & 0.32 & 0.22 & 0.37 & 0.25 \\
LSS $t_1$ quantile(0.95,$H_{\mathrm{E}}$) & 1 & 0.53 & 0.67 & 0.67 & 0.53 & 0.67 & 0.54 \\
LSS $t_2$ quantile(0.95,$H_{\mathrm{E}}$) & 0.56 & 0.53 & 0.53 & 0.53 & 0.53 & 0.5 & 0.53 \\
WSS quantile(0.95,$H_{\mathrm{E}}$) & 0.5 & 0.41 & 0.4 & 0.39 & 0.35 & 0.3 & 0.29 \\
WSS quantile(0.95, $D_{\mathrm{J}}$) & 0.49 & 0.32 & 0.12 & 0.11 & 0.06 & 0.06 & 0.07 \\
WSS quantile(0.95, $F_{\mathrm{ST}}$) & 0.64 & 0.51 & 0.25 & 0.25 & 0.15 & 0.16 & 0.18 \\
WSS quantile(0.95, $S$) & 14 & 18 & 16 & 18 & 20 & 30 & 27 \\
WSS quantile(0.95, $\theta_{\mathrm{W}}$) & 4.64 & 4.82 & 5.03 & 5.42 & 5.64 & 8.46 & 7.61 \\
WSS quantile(0.95, $\pi$) & 5.58 & 5.57 & 4.77 & 5.27 & 5.18 & 6.89 & 5.77 \\
WSS quantile(0.95, $D$) & 1.82 & 1.63 & 0.96 & 1 & 0.8 & 0.17 & 0.06 \\
WSS quantile(0.95, $D_{\mathrm{a}}$) & 0.38 & 0.23 & 0.08 & 0.08 & 0.04 & 0.04 & 0.05 \\
WSS $t_1$ quantile(0.95,$H_{\mathrm{E}}$) & 0.47 & 0.41 & 0.53 & 0.5 & 0.39 & 0.35 & 0.28 \\
WSS $t_2$ quantile(0.95,$H_{\mathrm{E}}$) & 0.53 & 0.37 & 0.42 & 0.4 & 0.36 & 0.3 & 0.32 \\
WSS $t_1$ quantile(0.95, $S$) & 4 & 15 & 10 & 11 & 16 & 13 & 13 \\
WSS $t_2$ quantile(0.95, $S$) & 13 & 15 & 14 & 17 & 15 & 28 & 24 \\
WSS $t_1$ quantile(0.95, $\theta_{\mathrm{W}}$) & 2.67 & 4.97 & 5.45 & 6 & 5.66 & 7.09 & 4.9 \\
WSS $t_2$ quantile(0.95, $\theta_{\mathrm{W}}$) & 4.6 & 4.97 & 4.95 & 5.63 & 5.3 & 8.44 & 7.95 \\
WSS $t_1$ quantile(0.95, $\pi$) & 2.67 & 5.2 & 5.5 & 6.17 & 5.62 & 7.17 & 4.66 \\
WSS $t_2$ quantile(0.95, $\pi$) & 5.6 & 4.67 & 4.8 & 5.3 & 5.09 & 7.01 & 6.56 \\
WSS $t_1$ quantile(0.95, $D$) & 3.19 & 1.45 & 1.63 & 1.63 & 1 & 1.09 & 1.01 \\
WSS $t_2$ quantile(0.95, $D$) & 2.09 & 1.14 & 1.05 & 1.06 & 1.05 & 0.3 & 0.38 \\

\end{longtable}
\end{tiny}







% Add all supplementary figures here:
\newpage

% Figure Supplementary 1 - Prior distribution of the PODs
\begin{figure}[ht]
 \centering
 \includegraphics[width=0.85\textwidth]{Figures/FigureS1_parameters_histograms.pdf}
 \small\caption{\textbf{Prior distribution of the simulation parameters, latent variables informative about the adaptive history and demography inferred with the ABC-RF.} (a) the per-site mutation rate $\mu$, (b) per base recombination rate per generation $r$, (c) the population size $N$, (d) the probability of beneficial mutation $P_RP_B$, (e) mean of the gamma distribution $\gamma$, (f) the proportion of strongly selected mutations $P$, (g) average selection coefficients of mutations under strong selection $\bar{s}$, (h) the scale mutation rate of beneficial mutations $\theta_{\mathrm{b}}$, (i) the average substitution load $L$, (j) the effective population size $N_{\mathrm{e}}$, (k) the distance between the effective size and population size expressed in the ratio $N_{\mathrm{e}}/N$.}
 \label{fig:supple_pods_priors}
\end{figure}

% Figure Supplementary 2 - PODs OOB plots << selection >>
\begin{figure}[ht]
 \centering
 \includegraphics[width=1\textwidth]{Figures/FigureS2_oob_plots_selection.pdf}
 \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the prediction of model parameters informative about the number of beneficial mutations and their strength, the latent variables associated with these two parameters, and variables informative about the adaptive dynamic.} (a) probability of a beneficial mutation to arise $P_RP_B$; (b) number of mutations under strong selection $P$; (c) mean of the gamma distribution $\gamma$; (d) mean of selection coefficients of mutations under strong selection $\bar{s}$; and (e) mean substitution load $L$. The bar plots on the left side of the scatterplots of $P$, $\bar{s}$, and $L$ correspond to the posterior estimates obtained for the simulations removed for training these parameters, as these simulations had zero as true values. MSE and $R^2$ estimates of b, d, and e corresponded to non-neutral simulations. See Table S2, Supplementary results, for neutral simulations MSE and bias.}
 \label{fig:supple_oob_sel}
\end{figure}

% Figure Supplementary 3 - PODs OOB plots << demography >>
\begin{figure}[ht]
 \centering
 \includegraphics[width=1\textwidth]{Figures/FigureS3_oob_plots_demography.pdf}
 \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the model parameters and latent variables informative about demography.} (a) mutation rate per generation $\mu$; (b) per base recombination rate per generation $r$; and (c) the ratio between the effective size and the population census size $N_{\mathrm{e}}/N$.}\label{fig:supple_oob_demo}
\end{figure}

% Figure Supplementary 4 [NEW] - Example of recombination rates
\begin{figure}[ht]
 \centering
 \includegraphics[width=1\textwidth]{Figures/FigureS4_example_variable_recomb.pdf}
 \small\caption{\textbf{How the recombination rate per base pair per generation $r$ varies across the genome.} Simulation for model training had constant $r$ (horizontal line in dark blue). Simulations used to test the ABC-RF robustness to heterogenous recombination rates could have different values of $r$ for the recombination map. This figure depicts one example of genome-wide $r$ for each type of simulation.}
 \label{fig:supple_example_recomb}
\end{figure}

% Figure Supplementary 5 - Expected vs. Predicted for simulations with variable recombination rate
\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS5_posterior_plots_recombination.pdf}
 \small\caption{\textbf{Expected versus posterior estimates plots for PODs with variable genome-wide recombination rates.} (a) $\theta_{\mathrm{b}}$, (b) $N$, (c) $N_{\mathrm{e}}$, (d) $F_{\mathrm{ST}}-N_{\mathrm{e}}$, (e) $P_RP_B$, (f) $P$, (g) $\gamma$, (h) $\bar{s}$, (i) $L$, (j) $\mu$, (k) $r$, and (l) $N_{\mathrm{e}}/N$. MSE and $R^2$ estimates of f, h, and i corresponded to non-neutral PODs. See Table S2, Supplementary results, for neutral PODs MSE and bias.}
 \label{fig:supple_pods_variable_recomb}
\end{figure}

% Figure Supplementary 6 - Variable importance plots - PODs selection parameters
\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS6_varplot_selection.pdf}
 \small\caption{\textbf{Variable importance plots of the trained ABC-RF for parameters informative about the adaptive history.} (a) The probability of beneficial mutation $P_RP_B$, (b) the proportion of strongly selected mutations $P$, (c) mean of the gamma distribution $\gamma$, (d) average selection coefficients of mutations under strong selection $\bar{s}$, (e) the scale mutation rate of selected mutations $\theta_{\mathrm{b}}$, and (f) the average substitution load $L$.}
 \label{fig:supple_pods_varplots_sel}
\end{figure}

% Figure Supplementary 7 - Variable importance plots - PODs demography parameters
\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS7_varplot_demography.pdf}
 \small\caption{\textbf{Variable importance plots of the trained ABC-RF for parameters informative about the adaptive history.} (a) the per-site mutation rate $\mu$, (b) per base recombination rate per generation $r$, (c) the population size $N$, (d) the effective population size $N_{\mathrm{e}}$, (e) the distance between the effective size and population size expressed in the ratio $N_{\mathrm{e}}/N$.}
 \label{fig:supple_pods_varplots_demo}
\end{figure}

% Figures Supplementary 8 to 14 Model check PCA.
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS8_model_check_Avalon.pdf}
  \small\caption{\textbf{Prior evaluation for the Avalon population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) if the model captured most of the targeted data features.}
  \label{fig:supple_model_check_avalon}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS9_model_check_Humboldt.pdf}
  \small\caption{\textbf{Prior evaluation for the Humboldt population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) if the model captured most of the targeted data features.}
  \label{fig:supple_model_check_humboldt}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS10_model_check_Davis.pdf}
  \small\caption{\textbf{Prior evaluation for the Davis population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) if the model captured most of the targeted data features.}
  \label{fig:supple_model_check_davis}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS11_model_check_Stanislaus.pdf}
  \small\caption{\textbf{Prior evaluation for the Stanislaus population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) if the model captured most of the targeted data features.}
  \label{fig:supple_model_check_stanislaus}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS12_model_check_Stebbins.pdf}
  \small\caption{\textbf{Prior evaluation for the Stebbins population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) if the model captured most of the targeted data features.}
  \label{fig:supple_model_check_stebbins}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS13_model_check_Riverside.pdf}
  \small\caption{\textbf{Prior evaluation for the Riverside population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) if the model captured most of the targeted data features.}
  \label{fig:supple_model_check_riverside}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS14_model_check_Placerita.pdf}
  \small\caption{\textbf{Prior evaluation for the Placerita population.} Projection of the targeted population principal components, computed from the summary statistics, into the cloud of the simulated data points obtained with principal component analysis (PCA). We see the targeted data point (in black) into the simulated data cloud of points (in grey) if the model captured most of the targeted data features.}
  \label{fig:supple_model_check_placerita}
\end{figure}

% Figure Supplementary 15, 16, and 17 - Feral Bees OOB plots << demography and selection >>
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS15_combined_plot_ne.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the population effective size $N_{\mathrm{e}}$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_NE}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS16_combined_plot_nc.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the population census size $N$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_N}
\end{figure}


\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS17_combined_plot_thetab.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the scaled mutation rate of beneficial mutations $\theta_{b}$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_thetab}
\end{figure}


% Figure Supplementary 18 to 22 - Feral Bees OOB plots << selection extra >>
\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS18_combined_plot_prps.pdf}
 \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the probability of beneficial mutation $P_RP_B$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
 \label{fig:supple_feralbee_prpb}
\end{figure}

\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS19_combined_plot_pstrong.pdf}
 \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the proportion of strongly selected mutations $P$ for pairs temporal samples of \textit{A. mellifera} feral populations.} Stacked points on the far left of each scatter-plot correspond to posterior estimates of neutral simulations, where $P = 0$. MSE and $R^2$ estimates corresponded to non-neutral simulations. See Table S2, Supplementary results, for neutral simulations MSE and bias.}
 \label{fig:supple_feralbee_pstrong}
\end{figure}

\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS20_combined_plot_gamma.pdf}
 \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the mean of the gamma distribution $\gamma$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
 \label{fig:supple_feralbee_gammamean}
\end{figure}

\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS21_combined_plot_gammasel.pdf}
 \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the average selection coefficients of mutations under strong selection $\bar{s}$ for pairs of temporal samples of \textit{A. mellifera} feral populations.} Stacked points on the far left of each scatter-plot correspond to posterior estimates of neutral simulations, where $\bar{s} = 0$. MSE and $R^2$ estimates corresponded to non-neutral simulations. See Table S2, Supplementary results, for neutral simulations MSE and bias.}
 \label{fig:supple_feralbee_gammaselmean}
\end{figure}

\begin{figure}[ht]
 \centering
 \includegraphics[width=0.95\textwidth]{Figures/FigureS22_combined_plot_load.pdf}
 \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the mean substitution load $L$ for pairs of temporal samples \textit{A. mellifera} feral populations.} Stacked points on the far left of each scatter-plot correspond to posterior estimates of neutral simulations, where $L = 0$. MSE and $R^2$ estimates corresponded to non-neutral simulations. See Table S2, Supplementary results, for neutral simulations MSE and bias.}
 \label{fig:supple_feralbee_load}
\end{figure}

% Figure Supplementary 23 to 25 - Feral Bees OOB plots << demography extra >>
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS23_combined_plot_mu.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for the mutation rate per generation $\mu$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_mu}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS24_combined_plot_rr.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the per base recombination rate per generation $r$ for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_r}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS25_combined_plot_nenc.pdf}
  \small\caption{\textbf{Out-of-bag estimates of ABC-RF trained for predicting the ratio between the effective size and the population census size for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_nen}
\end{figure}

% Figure Supplementary 26 NE Expected versus FST-NE prediction of Feral Bee populations.
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS26_combined_plot_fstNE.pdf}
  \small\caption{\textbf{$\hat N_{\mathrm{e}}$ estimates from the temporal $F_{\mathrm{ST}}$ to compare with the ABC-RF -based $\hat N_{\mathrm{e}}$ estimates for pairs of temporal samples of \textit{A. mellifera} feral populations.}}
  \label{fig:supple_feralbee_fstne}
\end{figure}

% Figure Supplementary 27 and 28 - Feral Bees Posterior density plots << selection and demography extra >>
\begin{figure}[ht]
  \centering 
  \includegraphics[width=0.95\textwidth]{Figures/FigureS27_weighted_densityPlots_selection_feralbees.pdf}
  \small\caption{\textbf{Posterior distributions of model parameters and a latent variable informative about selection for all feral \textit{A. mellifera} populations.} (a) probability of a beneficial mutation to arise $P_RP_B$; (b) number of mutations under strong selection $P$; (c)  mean of the gamma distribution $\gamma$; (d) mean of selection coefficients of mutations under strong selection $\bar{s}$; and (e) mean substitution load $L$. Dashed and filled lines correspond to the prior and posterior distributions, respectively. See Table S3, Supplementary Results for mean estimates and 95\% credibility intervals.}
  \label{fig:supple_feralbee_densityselection}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=0.95\textwidth]{Figures/FigureS28_weighted_densityPlots_demography_feralbees.pdf}
  \small\caption{\textbf{Posterior distributions of model parameters and a latent variable informative about demography for all feral \textit{A. mellifera} populations.} (a) mutation rate per generation $\mu$; (b) per base recombination rate per generation $r$; and (c) the ratio between the effective size and the population census size $N_{\mathrm{e}}/N$. Dashed and filled lines correspond to the prior and posterior distributions, respectively. See Table S3, Supplementary Results for mean estimates and 95\% credibility intervals.}
  \label{fig:supple_feralbee_densitydemo}
\end{figure}

\end{document}
